src/charm.py

```diff
@@ -117,6 +123,55 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
+
+    def _enable_mjolnir(self) -> None:
+        """Enable mjolnir service."""
+        container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
+            return
+        self.model.unit.status = ops.MaintenanceStatus("Configuring Mjolnir")
+        try:
+            admin_username = token_hex(16)
```

Comment:
Is 16 here the recommended value like described in [here](https://docs.python.org/3/library/secrets.html) - perhaps put a comment on the line?

Reply:
@gregory-schiano mentioned in one of his comments that we should consider creating an admin user systematically. At first I thought it could be temporary (I didn't want to add the complexity to keep a user/password) but then I realized that the access token created by the temporary user was deleted as well, making Mjolnir throw an error.

Reply:
I'll mark this one and the next regarding admin user in favor of this comment https://github.com/canonical/synapse-operator/pull/30#discussion_r1315842557 so we can discuss there the best approach. But feel free to reopen-it.